packages/expo/cli/utils/analytics/rudderstackClient.ts

```diff
@@ -0,0 +1,99 @@
+import RudderAnalytics from '@expo/rudder-sdk-node';
+import os from 'os';
+import { v4 as uuidv4 } from 'uuid';
+
+import { EXPO_LOCAL, EXPO_STAGING, EXPO_NO_TELEMETRY } from '../env';
+import UserSettings from '../user/UserSettings';
+
+const PLATFORM_TO_ANALYTICS_PLATFORM: { [platform: string]: string } = {
+  darwin: 'Mac',
+  win32: 'Windows',
+  linux: 'Linux',
+};
+
+let client: RudderAnalytics | null = null;
+let identified = false;
+let identifyData: {
+  userId: string;
+  deviceId: string;
+  traits: Record<string, any>;
+} | null = null;
+
+function getClient(): RudderAnalytics {
+  if (client) {
+    return client;
+  }
+
+  client = new RudderAnalytics(
+    EXPO_STAGING || EXPO_LOCAL ? '1wpX20Da4ltFGSXbPFYUL00Chb7' : '1wpXLFxmujq86etH6G6cc90hPcC',
+    'https://cdp.expo.dev/v1/batch',
+    {
+      flushInterval: 300,
+    }
+  );
+
+  // Install flush on exit...
+  process.on('SIGINT', () => client?.flush?.());
+  process.on('SIGTERM', () => client?.flush?.());
+
+  return client;
+}
+
+export async function setUserDataAsync(userId: string, traits: Record<string, any>): Promise<void> {
+  if (EXPO_NO_TELEMETRY) {
+    return;
+  }
+  const savedDeviceId = await UserSettings.getAsync('analyticsDeviceId', null);
+  const deviceId = savedDeviceId ?? uuidv4();
+  if (!savedDeviceId) {
+    await UserSettings.setAsync('analyticsDeviceId', deviceId);
+  }
+
+  identifyData = {
+    userId,
+    deviceId,
+    traits,
+  };
+
+  ensureIdentified();
+}
+
+export function logEvent(name: string, properties: Record<string, any> = {}): void {
```

Comment:
```suggestion

export enum AnalyticsEvent {
  ACTION = 'action', // generic event type which is used to determine the 'daily active user' stat, include an `action: expo ${subcommand}` property inside of the event properties object
}

export function logEvent(event: AnalyticsEvent, properties: Record<string, any> = {}): void {
```

Reply:
I chose a string union because it's lighter weight when transpiled down to JS. This code won't ship with types and users shouldn't import the code directly, so there's no need to support environments that don't have strong autocomplete (i.e. environments where enums would make dx nicer).